crates/ibc/src/core/ics03_connection/handler/conn_open_try.rs

```diff
@@ -4,117 +4,123 @@ use crate::core::ics03_connection::connection::{ConnectionEnd, Counterparty, Sta
 
-    // If a consensus proof is present, check that the consensus height (for
-    // client proof) in the message is not too advanced nor too old.
-    if let Some(consensus_height) = msg.consensus_height() {
-        check_client_consensus_height(ctx, consensus_height)?;
-    }
+    let connection_id_counter = ctx.connection_counter()?;
+    let self_connection_id = ConnectionId::new(connection_id_counter);
 
-    // Unwrap the old connection end (if any) and its identifier.
-    let (mut new_connection_end, conn_id) = match &msg.previous_connection_id {
-        // A connection with this id should already exist. Search & validate.
-        Some(prev_id) => {
-            let old_connection_end = ctx.connection_end(prev_id)?;
+    ///////////////////////////////////////////////////////////
+    // validate_self_client() verification goes here
+    // See [issue](https://github.com/cosmos/ibc-rs/issues/162)
+    ///////////////////////////////////////////////////////////
 
-            // Validate that existing connection end matches with the one we're trying to establish.
-            if old_connection_end.state_matches(&State::Init)
-                && old_connection_end.counterparty_matches(&msg.counterparty)
-                && old_connection_end.client_id_matches(&msg.client_id)
-                && old_connection_end.delay_period() == msg.delay_period
-            {
-                // A ConnectionEnd already exists and all validation passed.
-                output.log(format!(
-                    "success: `previous_connection_id` {} validation passed",
-                    prev_id
-                ));
-                Ok((old_connection_end, prev_id.clone()))
-            } else {
-                // A ConnectionEnd already exists and validation failed.
-                Err(Error::connection_mismatch(prev_id.clone()))
-            }
-        }
-        // No prev. connection id was supplied, create a new connection end and conn id.
-        None => {
-            // Build a new connection end as well as an identifier.
-            let conn_end = ConnectionEnd::new(
-                State::Init,
-                msg.client_id.clone(),
-                msg.counterparty.clone(),
-                msg.counterparty_versions.clone(),
-                msg.delay_period,
-            );
-            let id_counter = ctx.connection_counter()?;
-            let conn_id = ConnectionId::new(id_counter);
+    if msg.consensus_height > ctx.host_current_height() {
+        // Fail if the consensus height is too advanced.
+        return Err(Error::invalid_consensus_height(
+            msg.consensus_height,
+            ctx.host_current_height(),
+        ));
+    }
 
-            output.log(format!(
-                "success: new connection end and identifier {} generated",
-                conn_id
-            ));
-            Ok((conn_end, conn_id))
-        }
-    }?;
+    if msg.consensus_height < ctx.host_oldest_height() {
```

Comment:
I think we can remove this as suggested here -> https://github.com/cosmos/ibc-rs/issues/167#issuecomment-1275119757

Reply:
Agreed [25e58a6](https://github.com/cosmos/ibc-rs/pull/170/commits/25e58a6f215f380d405f05f2c792fb71cf81bf42)

Almost marked `host_oldest_height()` deprecated